lib/tether/tether_linux.go

```diff
@@ -140,9 +140,8 @@ func (t *tether) stopReaper() {
 	signal.Reset(syscall.SIGCHLD)
 
 	// just closing the incoming channel is not going to stop the iteration
-	// so we use done channel to signal it
-	log.Debugf("Signalling the child reaper loop")
-	close(t.done)
+	// so we use the context cancellation to signal it
+	t.cancel()
```

Comment:
Not sure I understood. Can you elaborate on why do you suggest having different signaling mechanisms?